### ISSUE TITLE
Local Backend attempt after server shutdown

### ISSUE THREAD

**Issue by thomas-br** [LFE89/nello_one_without_cloud#1]
Hi @LFE89,

thanks first of all for your great RE work on the nello. Sadly after the server shutdown, it is currently not possible any longer to use the nello devices. I would like to dig deeper into the topic of a local nello backend.

After going through your documentation my current understanding is, that we don't now the payload details of the mapping phase after resetting / initial setup, correct?

Do you still have the recorded messages between the backend and the nello device to continue reverse engineering on this?
It would be awesomely insane if we can reproduce a mapping, that the nello device accepts the local backend.

Best

**Comment by OnkelKeule** [comment]
Same here... looks like their domain transfer went wrong. I'm looking into all sorts of options. I'd love to get the Nello Hardware going ([this](https://github.com/atc1441/TCSintercomArduino) seems rather complex and would be my last resort).

I have gotten the Nello one to talk to my local MQTT server. Receiving notifications in case the door bell rings works fine. I wasn't able to trigger the door buzzer though. Systematically sending the right commands while at the same time forcing the Nello to reconnect appears to be not feasible in my setup... 

So probably the next step is to try to find out the encryption mechanism... Any ideas anyone?

**Comment by Atanasovgoran** [comment]
I wonder if it would be easier to simply write a custom firmware for the nello. It would be alot of work to write the code to communicate with the doorbell initially,but it would have the positive of being able to support new devices. Unfortunately my nello is not working, it's missing a part, (i think it's the voltage reg, but i don't know the part number) so I can't try reverse engineering the actual firmware.

**Comment by Atanasovgoran** [comment]
because i wouldn't be able to test it i mean

**Comment by LFE89** [comment]
Hey colleagues,

@thomas-br, your assumption is correct. I'll have a look on that and'll also check if we can figure out a way to RE the mapping.

@OnkelKeule, good to hear that the bell notification solution does work for you. 
Yeah, there are a lot of setups and reasons, where and why you don't want to force nello to reconnect. I see two further opps: either it possible to decipher and decrypt all the messages by analyzing the traffic (even by replaying the traffic...) or it is possible to reverse engineer the FW, in order to know how the system behaves correctly - as @Atanasovgoran also said.

I use my solution (the full BE impl.) since march this year and it works quite well. But sometimes, I have to admit, waiting for 10-12 seconds is a bit annoying ;-).

Maybe I can provide you another hint I figured out in march, but did not make public yet.
Sclak prob. did move all user accounts from nello to its system (at least my account) - without even asking me for my user account permissions - that is really really sad..
I was able get into the sclak system by simply "trying to reset" my password via: https://admin.sclak.com/#/forgot and they provided me all the information I needed to login.
Having said that, you're then able to retrieve a valid Access Token to their APIs via 

`[POST] https://api.sclak.com/auth_tokens (body: something like {"username": "myuser@mail.com", "password": "myPw"}).`

Afterwards you can call the Firmware API (all their Firmwares are listed there and can be downloaded):
https://api.sclak.com/firmwares

From my RE tasks back in march, I think that the Nello device firmwares are already listed there (just my guess).
Firmware Ids: 39, 40, 41 ,42 (all the ULock stuff things).
But unfortunately, these four firmwares are just set up as a placeholer without any data in it.
Checksum = d41d8cd98f00b204e9800998ecf8427e (MD5 Hash of an empty string).

Maybe we've to wait... I don't know. 
Maybe it will help you guys with your further analysis.

```
{
            "id": "42",
            "version": "1.05",
            "checksum": "d41d8cd98f00b204e9800998ecf8427e",
            "notes": "Ulock 1.05",
            "peripheral_type_id": "2",
            "peripheral_type_code": "ulock",
            "beta": "0",
            "support_pin": "0",
            "support_upgrade": "0",
            "support_battery": "0",
            "upgrade_mandatory": "0",
            "insert_time": "1444063305",
            "edit_time": "1444063305"
        },
        {
            "id": "41",
            "version": "1.04",
            "checksum": "d41d8cd98f00b204e9800998ecf8427e",
            "notes": "Ulock 1.04",
            "peripheral_type_id": "2",
            "peripheral_type_code": "ulock",
            "beta": "0",
            "support_pin": "0",
            "support_upgrade": "0",
            "support_battery": "0",
            "upgrade_mandatory": "0",
            "insert_time": "1444063305",
            "edit_time": "1444063305"
        },
        {
            "id": "40",
            "version": "1.02",
            "checksum": "d41d8cd98f00b204e9800998ecf8427e",
            "notes": "Ulock 1.02",
            "peripheral_type_id": "2",
            "peripheral_type_code": "ulock",
            "beta": "0",
            "support_pin": "0",
            "support_upgrade": "0",
            "support_battery": "0",
            "upgrade_mandatory": "0",
            "insert_time": "1444063305",
            "edit_time": "1444063305"
        },
        {
            "id": "39",
            "version": "1.0",
            "checksum": "d41d8cd98f00b204e9800998ecf8427e",
            "notes": "Ulock 1.0",
            "peripheral_type_id": "2",
            "peripheral_type_code": "ulock",
            "beta": "0",
            "support_pin": "0",
            "support_upgrade": "0",
            "support_battery": "0",
            "upgrade_mandatory": "0",
            "insert_time": "1444063305",
            "edit_time": "1444063305"
        },
```


Cheers Lars

**Comment by Atanasovgoran** [comment]
That's interesting. Can anyone post a picture of the PCB, it may help me find the broken component and fix it. I could then tryand helpwiththe reverse engineering

**Comment by OnkelKeule** [comment]
@Atanasovgoran - here's a picture of the front (took it a while ago... hope it helps):
![grafik](https://user-images.githubusercontent.com/26536207/86121871-161dad00-bad7-11ea-8306-25f539bfa42a.png)

@LFE89 - I just tried to reset my pw at sclak and reveived an error message. Seems like my account was not transferred...

**Comment by Atanasovgoran** [comment]
yes i can see the part i am missing, seems to be a battery or a supper cap, will have to search for it when i am home

**Comment by LFE89** [comment]
@OnkelKeule , that is quite interesting. Why did they then migrate my account...
I can even see that they did that prob. already on 19.02.2020.

My tests through the inofficial Sclak API back in march were all successful, because they either redirected the requests to the origin BE or already owned the infrastructure....

@OnkelKeule Did you try to create an account with your nello mail address? I remember that I got a strange error message, that it is not possible by using that mail address. Afterwards I tried the reset password thing...

![01_SCLAK](https://user-images.githubusercontent.com/60022287/86123647-2b480b00-bada-11ea-9c5a-c41043195980.png)
![02_SCLAK](https://user-images.githubusercontent.com/60022287/86123664-300cbf00-bada-11ea-8f3d-846c85c0810c.png)
![03_SCLAK](https://user-images.githubusercontent.com/60022287/86123670-326f1900-bada-11ea-803c-546cb5b4e6e7.png)

**Comment by OnkelKeule** [comment]
@LFE89 - seems to be Italian style. I'm using gmail's ["+"](https://gmail.googleblog.com/2008/03/2-hidden-ways-to-get-more-from-your.html), which obviously Sclak can't handle. So was able to register a new account (which I won't be able to use, as I can't very my email adress) but Sclak got it wrong:
![grafik](https://user-images.githubusercontent.com/26536207/86125564-2b95d580-badd-11ea-8269-ca4bb12abf42.png)

**Comment by secuninja** [comment]
Same here. Can login to Sclak after password reset but Nello is not working. DNS request to live-mqtt.nello.io results in NXDOMAIN. Could test customized firmware

**Comment by thomas-br** [comment]
Does anybody know a source of the existing nello firmwares other than maybe in the future at the sclak firmware api?

**Comment by AfromD** [comment]
I guess when I didn't get this set up while the service was working (to capture packets), I am out of luck now until the official service comes back up?

**Comment by LFE89** [comment]
@AfromD unfortunately yes. As long as we are not able to create new valid „test“-messages by our own, you need at least one recorded message (to support door open commands)... Maybe, it should be checked if „test“-messages from other nello devices may work for other devices as well. I am not convinced, because they prob. include unique device identifier or signature information to prohibit that, but maybe it is worth trying.

**Comment by AfromD** [comment]
> 
> 
> Maybe, it should be checked if „test“-messages from other nello devices may work for other devices as well. I am not convinced, because they prob. include unique device identifier or signature information to prohibit that, but maybe it is worth trying.

@LFE89 Wouldn't that be a huge security hole? A local attacker would use ARP spoofing to direct Nuki to his server, and then just serve up prerecorded packets from anywhere to unlock a door....
I might want to get the Nuki opener just as well if that proves to be true...

**Comment by LFE89** [comment]
@AfromD the security issue already exists - also in another manner. In the „old“ cloud environment. Any nello user, or let’s say any user who has access to a valid nello id (just the id), could connect to the mqtt broker and read all the messages, of each nello device. Including the test messages. And the fact, that there is prob. a misconfigured BE or a bug in the fw (still needs to be proven) which lead in my online scenario test to 15-20 minutes reconnects, let attacker to not wait long time to capture the right packets. Only the restricted write access to the specific topic prohibited further access. Employees prob. had access to that topic and could have control all locks within that attack scenario, without knowing any users sensitive data / passwords .

**Comment by LFE89** [comment]
And yes. By doing a dns/arp spoofing attack, an attacker could do that locally. There is no way to prevent that with the current firmware.

The attacker just had to do the following things:
- Spoofing attack to get the nello id
- End spoofing
- Connect to cloud nello mqtt Broker with another valid Id
- Read messages from the specific nello device topic
- Spoofing attack with valid test message / security bypass sequence.

**Comment by OnkelKeule** [comment]
One thing that I don't understand:
with Nello working I always received a Push notification when I pressed the buzzer to open the door.
With Nello using my local mqtt server, I don't see any communication whe I press the buzzer. 
How's that explained? Confuses me...

**Comment by secuninja** [comment]
Since the migration from nello to Sclak I never got any ring notification anymore

**Comment by warlo** [comment]
Looks like live-mqtt.nello.io is responding now. Haven't tried it out yet, but might be viable to catch the message again.

**Comment by thiroxx** [comment]
@LFE89 Looks like im migrated too, on Feb 19. 
Have you seen this? [FHEM](https://forum.fhem.de/index.php/topic,75127.0.html?PHPSESSID=be1kj3bbua1sldirut54k2o8nt)
I have used this a long time to get Nello into HomeKit and they got the whole communication working.

**Comment by secuninja** [comment]
> Looks like live-mqtt.nello.io is responding now. Haven't tried it out yet, but might be viable to catch the message again.

the error message is gone but neither unlocking nor ring-push is working

**Comment by OnkelKeule** [comment]
> Looks like live-mqtt.nello.io is responding now. Haven't tried it out yet, but might be viable to catch the message again.
> the error message is gone but neither unlocking nor ring-push is working

Same here

**Comment by LFE89** [comment]
> @LFE89 Looks like im migrated too, on Feb 19.
> Have you seen this? [FHEM](https://forum.fhem.de/index.php/topic,75127.0.html?PHPSESSID=be1kj3bbua1sldirut54k2o8nt)
> I have used this a long time to get Nello into HomeKit and they got the whole communication working.

There are a lot of such solutions in place. But all of them make use of either the official (a long time ago you just could request an official API account) or inofficial API (the API the original app used) and therefore have a dependency on the public cloud backend.

**Comment by LFE89** [comment]
> > Looks like live-mqtt.nello.io is responding now. Haven't tried it out yet, but might be viable to catch the message again.
> > the error message is gone but neither unlocking nor ring-push is working
> 
> Same here

Hope you can record your relevant mqtt messages.

**Comment by secuninja** [comment]
to be honest, i would be more interested in a custom firmware. nello cloud api failed often enough now...

**Comment by LFE89** [comment]
@secuninja totally agree. But you can try my non-cloud solution (without nello cloud api dependencies) as long as there is no other firmware available.

**Comment by alexanderbjelos** [comment]
Hello,

for now I can use my migrated Sclak account by reset my password. Furher I can log in into the Sclak App, but my Nello is still not working at the moment.

![image](https://user-images.githubusercontent.com/22572905/86887522-f6027500-c0f8-11ea-9cd7-ae86e02bfa4a.png)

Alex

**Comment by secuninja** [comment]
ditto... not working at all

**Comment by thiroxx** [comment]
Looks like its working again. The Device became today reponsive again in Homebridge. Even the Api with token creation is up and running again.

**Comment by alexanderbjelos** [comment]
Yes, it works. Thanks for the information @thiroxx  :-)

**Comment by LFE89** [comment]
But please keep in mind, that the security issue still persists ([4.3. Control nello (the attack)](https://github.com/LFE89/nello_one_without_cloud#43-control-nello-the-attack)) in the current SCLAK cloud & FW solution.

IMHO it is important to know about the security issue, then it is up to the customers if they are willing to take the risk explicitly or not.
Unfortunately neither Nello nor Sclak commented on the issue I reported back in february. So take care...

**Comment by thomas-br** [comment]
Was anybody able to get further insights on the traffic exchanged during the setup phase?

**Comment by thomas-br** [comment]
One update from my end. I recorded again some reset / setup flows as tcpdump. Basically no new insights besides what is already shared by @LFE89 in his Readme. 

Interesting are indeed the encrypted messages. I am wondering whether it is based on a static key for all nellos, a static one put into the hardware during manufacturing or exchanged / constructed out of the blinking which syncs the wifi information.

**Comment by thomas-br** [comment]
Some more thoughts... I opened also my nello for find a way to get the encryption key which seems to be put into the device during manufacturing (that at least what the nello product page says) and is used for the AES-256 encryption. I mean sometimes such keys are even included as a QR (or similar) code onto the board [[1]](https://insinuator.net/2016/04/discover-the-unknown-analyzing-an-iot-device/). Besides the MAC-QR I found another one on the back of my nello IC-board. Unfortunately it only contains a 10-digit number (starting with `994...`)

I though dumping the binary data from the flash chip could also be worth a try [[2]](https://nvisium.com/blog/2019/08/07/extracting-firmware-from-iot-devices.html) [[3]](https://www.oreilly.com/library/view/iot-penetration-testing/9781787280571/1ac59502-2a43-4625-abae-d3e4710a88b5.xhtml)
I was not able to locate an isolated flash chip where we can hook a SOIC clip to. As the nellos use a ATWINC1500/ATWINC1510 which also contains afaik 8MB flash and OTA support. So the firmware might be contained in here, or what do you think? Or is it really just for the Wifi connectivity?

Does anybody has an idea how we could dump a `.bin`?

**Comment by sistein** [comment]
Thanks for your work. Just found it now as it seems that the original server is down again...

**Comment by alexanderbjelos** [comment]
> Thanks for your work. Just found it now as it seems that the original server is down again...

Since yesterday nello hasn’t a connection to the Sclak server. Tomorrow I will try to setup your solution without cloud. Thank you @LFE89

**Comment by secuninja** [comment]
Same here... have you been successful on recording the test-message so far?

**Comment by Hypfer** [comment]
@thomas-br I'd assume that the firmware is stored inside the nordic BLE uC found on the back of the board

It wouldn't surprise me if wifi was actually just an afterthought and therefore glued on top of the finished product by adding another module.
There may or may not be protections against dumping the flash of that uC but at least there are testpoints available for all the required connections.

To try this, you will need an SWD programmer/debugger and rather precise soldering skills because those testpoints aren't very large.
You could also try using one of these conductive needle things, but keep in mind that the required testpoints are spread on both sides of the PCB

**Comment by thomas-br** [comment]
@Hypfer either on the BT uC or also possible (and actually my initial thought) on the Wifi uC (ATWINC1510). Based on the data sheet it offers enough storage for the firmware and even features like OTA upgrades which they might planned to use in the future. 

I read firmware dumping might be possible over the SPI Interface here. But as you said, first challenge would be soldering to the pins which are in fact not huge 


How come you rather think the firmware is on the BT chip instead of the ATWINC which seems way more "mature" based on my readings

**Comment by thomas-br** [comment]
> Same here... have you been successful on recording the test-message so far?

As the mqtt messages (payloads) are encrypted, the first challenge in my eyes is more to get to the key of your own device to be able to understand what is going on

**Comment by Hypfer** [comment]
> How come you rather think the firmware is on the BT chip instead of the ATWINC which seems way more "mature" based on my readings

That's just a hunch after having seen other IoT stuff which was developed at a similar time. I think nordic has/had very easy to use devkits or something?
I also don't see any reason for why else there would be that additional BLE uC on the board if not because it actually contains the logic part.
iirc configuration was done via the wifi hotspot? It's been a while since I've tried that

**Comment by Hypfer** [comment]
See here:
https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF52-DK

An arduino-compatible breakout board. The software/docs are also quite nice

The chip on the back side of the nello PCB is labelled N52832 QFAAB0 1718GC

**Comment by cristianodri** [comment]
Hello, have you take a look at this repository: https://github.com/nello-io
There're drivers for the board and some documentation. Perhaps it is possible to reconfigure the nello someway?

**Comment by Robbilie** [comment]
Is it just me or is their api down again? ^^

**Comment by cristianodri** [comment]
yes, their API is down since last Friday

**Comment by Robbilie** [comment]
The mqtt server seems to be down on live-mqtt.nello.io…
Interestingly enough, there seems to be a node exporter running there:
http://live-mqtt.nello.io:9100/metrics
AWS Hosted Kubernetes node with an ubuntu vm

**Comment by M4D1NG3R** [comment]
I still don't get a response from the server. Is there any news regarding integration with sclak, e.g. new firmware?

It's really frustrating that I can't use the device any more.

**Comment by cristianodri** [comment]
Sclak filed for bankruptcy in October 2020. In mid January 2021 there's been an auction to sell the company but I don't know if someone bought it or not, so I don't think we should hold our breath in hopes to see the servers up again

**Comment by nox26** [comment]
Hey! 

Im not sure but the mqtt Server seams working again, at least ping work... can somebody confirm ... (can't Check it since the Nello app always says network error on the initial setup since I have reset my nello)

**Comment by 2h4u** [comment]
Hey, just checked but sadly for me it doesn't work 😕

**Comment by pattyland** [comment]
```
pattyland@air ~ % telnet mqtt.nello.io 1883
Trying 52.59.132.203...
Connected to mqtt.nello.io.
Escape character is '^]'.
Connection closed by foreign host.
pattyland@air ~ % telnet live-mqtt.nello.io 1883
Trying 3.120.140.153...
telnet: connect to address 3.120.140.153: Connection refused
telnet: Unable to connect to remote host
```
 I would say no. Afaik ping and other services but mqtt worked the whole time

**Comment by nox26** [comment]
Damn, I hoped it work, since I need a "Test topic" or my new nello is useless :/

**Comment by Hypfer** [comment]
IMO, the only reasonable thing to do now would be to try to glitch the nrf to bypass the read-out protection and take a look at its firmware, which seems to be surprisingly easy for people skilled with hardware.

https://github.com/atc1441/ESP32_nRF52_SWD

**Comment by Robbilie** [comment]
good night sweet prince, https://nello.io is down now aswell

**Comment by pattyland** [comment]
Even the API is down... I made an unofficial statuspage a few days ago: https://stats.uptimerobot.com/5wQ9jh0mm1

**Comment by sistein** [comment]
SCLAK scheint nun jemand anderem zu gehören...

UDINO S.R.L., with registered office in via Amedeo Avogadro, 24 10121 Torino, VAT no. 11989000010

Eventuell wird da aktuell umgebaut...

**Comment by 1Joe1** [comment]
Flashing nello one with tasmota isn't possible, is it?

**Comment by innotip-com** [comment]
Folks, I think that our last chance to use our nello is to get the domain upon expiry, hoping no one will take it before :D "2022-04-07" ;) I guess that if some of you are able to kind of "simulate" the nello servers, we could make it work, couldn't we ?

**Comment by pattyland** [comment]
@innotip-com With a local DNS server it is no problem to connect Nello to a running MQTT server. The challenge is the logic, which is described in the readme of this repository: https://github.com/LFE89/nello_one_without_cloud/blob/master/README.md

**Comment by My1GH** [comment]
Can I with a local mqtt server and DNS overwrite, detect rings and issue and "open door" command?
I guess the phone app is totally useless now.

**Comment by bobo-in** [comment]
Hey Lars, thanks for your work!

Is it possible to receive only ring bell notifications without the message to the test topic from nello's MQTT server? 
live-mqtt.nello.io is online but it does not send messages to the test-topic when establishing a connection :(

Would the security bypass also be possible with your recorded message to the test-topic? Or do I need the message that was sent to my device ?

**Comment by XDjackieXD** [comment]
I'm currently in the process of reverse engineering the device firmware.
Apparently the firmware is compiled again for each device as the device IDs are compiled into the topic strings and such (could also be templating in the binary but I highly doubt that).
I'll share more info once I know a bit more (it has been a long time since I last reversed an embedded firmware...)

**Comment by 1Joe1** [comment]
@XDjackieXD How are you doing? Any news or progress? Anything one can do to support? I am counting on you 😉. Would be really awesome to reactivate my nello. Couldn't find a replacement so far, that is equally suited.

**Comment by XDjackieXD** [comment]
I've identified some standard-lib functions and some mqtt related functions but it all being non-blocking code with pseudothreading in a RTOS makes it annoying to reverse-engineer because almost everything has some global state stored in RAM.
I've also found some functions related to the "blink code transfer" of the wifi password but there's not really anything interesting to be found there.

If someone wants to also dump their firmware: I was using [this ESP32 based glitcher project](https://github.com/atc1441/ESP32_nRF52_SWD) with a delay setting of 5290-5370.
You have to remove the two capacitors on the DEC1 pin and cut the trace of the wifi module's enable line between the microcontroller and its pull-down resistor (because the wifi module once enabled draws too much current and the glitch won't work reliably; see the bottom left of the picture below)

![image](https://user-images.githubusercontent.com/4963868/150120937-c498065a-2dd9-45a1-8f52-b17ac4e444ae.png)

As for reverse engineering I'm currently using Ghidra as it seems to work better for embedded arm firmware than radare2.

It'll take at least two more weeks until I can work some more on this because I have some upcoming exams in university which I have to focus on at the moment.

**Comment by XDjackieXD** [comment]
On another note: does someone have a valid mqtt message (for example a door message) sent from the nello server to their device and would share it's content with me? that would help checking if I'm on the right track

**Comment by sistein** [comment]
Why was this closed? Issue solved? ^^

**Comment by thomas-br** [comment]
> Why was this closed? Issue solved? ^^

I felt like this issue is not really the right place for further communication and exchange any longer.
Is issue is quite old, the thread is quite hard to follow from the start with multiple topics being discussed in parallel.

At least my hopes for the nello backend to function again in the cloud are quite low due to the bankruptcy of SCLAK.
In my mind, dedicated locations for further exchange would be a better fit... Whatever / wherever that is.

**Comment by XDjackieXD** [comment]
We could start a new issue for the reverse engineering but in my opinion GitHub issues is a good place to work on this as it can be read by anyone in the future as a rough documentation.
Moving this to some other chat would mean that many people would not get any idea of what's going on and the discussions will most likely be lost forever after a few weeks have passed

**Comment by ghost** [comment]
has there been any progress ?

**Comment by thomas-br** [comment]
Hi all,

I know it has been a long time and also sorry for being pessimistic in January for not being able to see high chances on this 😄. But I wanted to give a quick note that I was able to spend some time during the last months looking into the Nello Topic again. I was able to extract the contents of the flash, locate the device specific encryption key material & reverse engineer the command protocol between server & the nello.

As a small PoC I was able to patch the hostname in my firmware to have to talk to a minimal server that currently just keeps my nello "online" (for now due to time constraints) and at least sending Door Bell Notifications to a Telegram Bot.
My personal plan is definitely to work on a server that comes close to feature parity of the original backend.
Goal would be to keep using the nello as door opener with its core capabilities – but I cannot say when I can spend how much time to further work on this.

Of course in general it would also be possible to open source such a project from the beginning on, where interested folks could even contribute. But I was not sure whether I am the only one currently still interested in keeping my nello away from being electronic waste. If there is general interest for such a project, you could maybe indicate it via emoji reactions

Screenshot Ring Notification of my PoC:
<img width="454" alt="image" src="https://user-images.githubusercontent.com/46727578/198526624-21996403-3524-4ddf-9a25-44bbda4f53bd.png">

**Comment by sistein** [comment]
Actually my Nello(s) are still installed. I did not find the time to build myself I solution (and I think Nello has had a lot of good design choices - for example the power source). 

So yes. Definately interested.

**Comment by innotip-com** [comment]
I’m definitely interested and would be fine helping with financing someone to work in this ?
Le 28 oct. 2022 à 09:19 +0200, thomas-br ***@***.***>, a écrit :

Hi all,

I know it has been a long time and also sorry for being pessimistic in Jan not being able to get any progress on this 😄. But I wanted to give a quick note that I was able to spend some time during the last months looking into the Nello Topic again. I was able to extract the contents of the flash, locate the device specific encryption key material & reverse engineer the command protocol between server & the nello.

As a small PoC I was able to patch the hostname in my firmware to have to talk to a minimal server that currently just keeps my nello "online" (for now due to time constraints) and at least sending Door Bell Notifications to a Telegram Bot.
My personal plan is definitely to work on a server that comes close to feature parity of the original backend.
Goal would be to keep using the nello as door opener with its core capabilities – but I cannot say when I can spend how much time to further work on this.

Of course in general it would also be possible to open source such a project from the beginning on, where interested folks could even contribute. But I was not sure whether I am the only one currently still interested in keeping my nello away from being electronic waste. If there is general interest for such a project, you could maybe indicate it via emoji reactions

Screenshot Ring Notification of my PoC:
[image]<https://user-images.githubusercontent.com/46727578/198526624-21996403-3524-4ddf-9a25-44bbda4f53bd.png>

**Comment by thomas-br** [comment]
I will start document my findings regarding the protocol as soon as possible.
Will let you know once I have a state that can be shared.
Sorry for the delay, I am currently pretty short on free-time.

**Comment by sistein** [comment]
I think we all know being short on free-time (or time at all).

**Comment by 1Joe1** [comment]
I ordered a Ring Intercom as a replacement for my bello just a few weeks ago, believing that your efforts and therefore my last hope got the nello had died. I received and installed it this weekend, but have to admit, that even though it is mostly doing, what it is supposed to, I liked the nello better and would love to put it back in place. It is in my eyes simply the smarter solution (no batteries, fits inside my intercom phone, ...). 
So any progress is highly appreciated!

**Comment by thomas-br** [comment]
I was able to find some time to write down at least the most important stuff. It is by far not complete, neither by what I have reverse engineered, nor by where there are potentials for research. Still will try to extend it with my knowledge about the system from time to time. Especially the information regarding firmware patching what I did to connect to another MQTT server is not yet in (will do).

https://github.com/thomas-br/open-nello

I think it would be best to further discuss technical details and your questions over there (in the issues I have created one for this). Will also try to push my poc for a open source server implementation over the winter holidays when I hopefully find some time to continue my work on it. (There is also another issue for discussions and coordination around an open-source server solution)
